rpc/src/transaction_status_service.rs

```diff
@@ -241,14 +253,17 @@ impl TransactionStatusService {
                 }
             }
             TransactionStatusMessage::Freeze(bank) => {
+                if !bank.is_frozen() {
+                    return Err(Error::NonFrozenBank(bank.slot()));
+                }
```

Comment:
This check prevents a similar regression in the future and fails loudly (bring the validator down). This means that this thread will now need to obtain the below lock:
https://github.com/anza-xyz/agave/blob/8e60bf233e3818744022dd7f2350d0788798551a/runtime/src/bank.rs#L756-L757
 This doesn't seem awful tho given that it is a `RwLock` and `Bank::freeze()` should be the only writer. Thus, lock contention shouldn't be much of a concern here, just the extra overhead of getting the read lock within `Bank::is_frozen()`:
 https://github.com/anza-xyz/agave/blob/8e60bf233e3818744022dd7f2350d0788798551a/runtime/src/bank.rs#L2087-L2089